### PR TITLE
[Devx: 177] Listing interface

### PR DESCRIPTION
**Pull request by sainivedh**
## What
- Centralised listing interface for all apps, inputs, datasets, models, workflows, modules ..etc
- Changed `base_auth` to a more appropriate `base_client` 
- Added support for dynamic logging levels

### CODE REVIEW

File: clarifai/client/lister.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    except ValueError:
+      datetime_obj = datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%SZ')
+
+    # Convert the datetime object to a Timestamp object
+    timestamp_obj = Timestamp()
+    timestamp_obj.FromDatetime(datetime_obj)
+
+    return timestamp_obj
+
+  def convert_keys_to_snake_case(self, old_dict: Dict[str, Any],
```
Comment (steph-en-m) [inline clarifai/client/lister.py:35]: Why is this needed?
I would think the integrity of the data fetched from the APIs must be maintained.

File: clarifai/client/app.py [hunk 2]
```diff
@@ -1,21 +1,31 @@
 from clarifai_grpc.grpc.api import resources_pb2, service_pb2  # noqa: F401
 
-from clarifai.client.base_auth import BaseAuth
+from clarifai.client.base_client import BaseClient
```
Comment (steph-en-m) [inline clarifai/client/app.py:3]: should `base_client.py` just be `base` since it lives under client already?

File: clarifai/client/lister.py (hunk 1 above)
Comment (sainivedh) [reply clarifai/client/lister.py:35]: This is needed for initialising/creating objects from respective resource protos.

The MultiOutputResponse returns response with keys as `userId`, `defaultLanguage` etc... but the protos expects all keys to be snake cased user_id, default_language

File: clarifai/client/lister.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    Yields:
+        dict: The next item in the listing.
+    """
+    page = 1
+    while True:
+      request_data['page'] = page
+      response = self._grpc_request(endpoint, proto_message(**request_data))
+      dict_response = MessageToDict(response)
+      if response.status.code != status_code_pb2.SUCCESS:
+        raise Exception(f"Listing failed with response {response!r}")
```
Comment (steph-en-m) [inline clarifai/client/lister.py:81]: what's `{response!r}` supposed to represent?

File: clarifai/client/lister.py (hunk 1 above)
Comment (steph-en-m) [reply clarifai/client/lister.py:35]: It looks like you do the convertion after retrieving data from the `List` endpoints

Comment (sainivedh) [reply clarifai/client/lister.py:35]: Yes, we get all responses from `List` endpoints and return as respective class object for each of the response

Comment (sainivedh) [reply clarifai/client/lister.py:35]: Yes. Only the keys are modified, not the values

Comment (steph-en-m) [reply clarifai/client/lister.py:35]: Got it

File: clarifai/client/lister.py (hunk 3 above)
Comment (sainivedh) [reply clarifai/client/lister.py:81]: The !r formatting specifier is used to represent the response object using its repr() method.

**Review by steph-en-m — APPROVED** [review]
LGTM
